libsrc/dataset.cpp

```diff
@@ -139,14 +151,20 @@ std::string Dataset::readHeader()
     return xml;
 }
 
-template <typename T> void Dataset::appendAcquisition(const Acquisition<T>& acq, int stream)
+/**********************************************************************************************************************/
+template <typename T> void Dataset::appendAcquisition(const Acquisition<T>& acq, StreamId stream)
 {
-    if (stream < 0) {
-        stream = ISMRMRD_MRACQUISITION_STREAM; // TODO: Temporary workaround.
-        //stream = acq.getStream();
+    if (stream == STREAM_NONE)
+    {
+      stream = STREAM_MRACQUISITION_DEFAULT;
```

Comment:
This is what I am talking about here. It is silly to have this kind of code when you could make it clear in the function signature with `StreamId stream = STREAM_MRACQUISITION_0`